src/main/scala/is/hail/utils/HashMethods.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.utils
+
+import org.apache.commons.math3.random.RandomDataGenerator
+
+object SimpleTabulationHash32 {
+  def apply(rand: RandomDataGenerator): SimpleTabulationHash32 = {
+    new SimpleTabulationHash32(Array.fill[Array[Int]](4)(Array.fill[Int](256)(rand.getRandomGenerator.nextInt())))
```

Comment:
Do you want a single array of size 1024 to avoid the double-indirection?

Reply:
I don't understand cache behavior well enough to know if it's better to have one large contiguous array, or a few smaller ones, but I think you're probably right.

Reply:
The cache behavior will be similar but slightly worse in the nested array case.  You need to cache the smaller array and the 256-element arrays may not be cache-line-aligned, so the cache won't be used quite so efficiently.

I think the bigger effect is you do an extra memory reference.